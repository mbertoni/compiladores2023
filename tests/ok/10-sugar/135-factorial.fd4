let rec fact (x:Nat) :Nat = ifz x 
                            then 1 
                            else x * (fact5 (x-1))