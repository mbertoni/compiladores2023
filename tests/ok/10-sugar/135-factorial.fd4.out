x = 1
y = 1
x1 = 1
cont = 0
x1 = 1
cont = 1
x = 2
y = 1
x1 = 2
cont = 0
x1 = 2
cont = 1
x1 = 2
cont = 2
x = 3
y = 2
x1 = 3
cont = 0
x1 = 3
cont = 1
x1 = 3
cont = 2
x1 = 3
cont = 3
x = 4
y = 6
x1 = 4
cont = 0
x1 = 4
cont = 1
x1 = 4
cont = 2
x1 = 4
cont = 3
x1 = 4
cont = 4
x = 5
y = 24
x1 = 5
cont = 0
x1 = 5
cont = 1
x1 = 5
cont = 2
x1 = 5
cont = 3
x1 = 5
cont = 4
x1 = 5
cont = 5
5! = 120
